NetTopologySuite/IO/ConfigurableBinaryReader.cs

```diff
@@ -0,0 +1,220 @@
+        /// </returns>
+        /// <exception cref="T:System.ObjectDisposedException">The stream is closed. </exception>
+        /// <exception cref="T:System.IO.IOException">An I/O error occurs. </exception>
+        /// <exception cref="T:System.IO.EndOfStreamException">The end of the stream is reached. </exception>
+        [CLSCompliant(false)]
+        public override ulong ReadUInt64()
+        {
+            var result = base.ReadUInt64();
+            return (EncodingType == ByteOrder.BigEndian)
+                ? BitTweaks.ReverseByteOrder(result)
+                : result;
+        }
+
+        /// <summary>
+        /// Reads a 4-byte floating point value from the current stream using the specified encoding
+        /// and advances the current position of the stream by four bytes.
+        /// </summary>
+        /// <returns>
+        /// A 4-byte floating point value read from the current stream.
+        /// </returns>
+        /// <exception cref="T:System.ObjectDisposedException">The stream is closed. </exception>
+        /// <exception cref="T:System.IO.IOException">An I/O error occurs. </exception>
+        /// <exception cref="T:System.IO.EndOfStreamException">The end of the stream is reached. </exception>
+        public override float ReadSingle()
+        {
+            var result = base.ReadSingle();
+            return (EncodingType == ByteOrder.BigEndian)
+                ? BitTweaks.ReverseByteOrder(result)
+                : result;
+        }
+
+        /// <summary>
+        /// Reads an 8-byte floating point value from the current stream using the specified encoding
+        /// and advances the current position of the stream by eight bytes.
+        /// </summary>
+        /// <returns>
+        /// An 8-byte floating point value read from the current stream.
+        /// </returns>
+        /// <exception cref="T:System.ObjectDisposedException">The stream is closed. </exception>
+        /// <exception cref="T:System.IO.IOException">An I/O error occurs. </exception>
+        /// <exception cref="T:System.IO.EndOfStreamException">The end of the stream is reached. </exception>
+        public override double ReadDouble()
+        {
+            var result = base.ReadDouble();
+            return (EncodingType == ByteOrder.BigEndian)
+                ? BitTweaks.ReverseByteOrder(result)
+                : result;
+        }
+
+        /// <summary>
+        /// Reads a string from the current stream. 
+        /// The string is prefixed with the length, encoded as an integer seven bits at a time.
+        /// </summary>
+        /// <returns>The string being read.</returns>
+        /// <exception cref="T:System.ObjectDisposedException">The stream is closed. </exception>
+        /// <exception cref="T:System.IO.IOException">An I/O error occurs. </exception>
+        /// <exception cref="T:System.IO.EndOfStreamException">The end of the stream is reached. </exception>
+        [Obsolete("Not implemented")]
```

Comment:
Can we also add `, error: true` to these two `[Obsolete]` attributes so it outright fails to compile if someone tries to use these?

Reply:
Obviously won't help with people who call these virtual methods virtually, but it's the best we can do...

Reply:
Why `[Obsolete]` in the first place, just call the base function if Endianess is little, otherwise throw NotSupportedException();
This is a tweak for our problem, not for the world.